src/license_manager_agent_ops.py

```diff
@@ -27,49 +28,77 @@ class LicenseManagerAgentOps:
     EPILOG_PATH = _VENV_DIR / "bin/slurmctld_epilog"
     _SLURM_USER = "slurm"
     _SLURM_GROUP = "slurm"
+    _LICENSE_MANAGER_USER = "license-manager"
+    _LICENSE_MANAGER_ACCOUNT = "license-manager"
+
 
     def __init__(self, charm):
         """Initialize license-manager-agent-ops."""
         self._charm = charm
 
     def setup_cache_dir(self):
         """Set up cache dir."""
-        CACHE_DIR = Path("/var/cache/license-manager")
-
         # Delete cache dir if it already exists
-        if CACHE_DIR.exists():
-            logger.debug(f"Clearing cache dir {CACHE_DIR.as_posix()}")
-            rmtree(CACHE_DIR, ignore_errors=True)
+        if self._CACHE_DIR.exists():
+            logger.debug(f"Clearing cache dir {self._CACHE_DIR.as_posix()}")
+            rmtree(self._CACHE_DIR, ignore_errors=True)
         else:
             logger.debug(
-                f"Tried to clean cache dir {CACHE_DIR.as_posix()}, but it does not exist"
+                f"Tried to clean cache dir {self._CACHE_DIR.as_posix()}, but it does not exist"
             )
 
         # Create a clean cache dir
-        logger.debug(f"Creating a clean cache dir {CACHE_DIR.as_posix()}")
-        CACHE_DIR.mkdir(parents=True)
-        chown(CACHE_DIR.as_posix(), self._SLURM_USER, self._SLURM_GROUP)
-        CACHE_DIR.chmod(0o700)
+        logger.debug(f"Creating a clean cache dir {self._CACHE_DIR.as_posix()}")
+        self._CACHE_DIR.mkdir(parents=True)
+        chown(self._CACHE_DIR.as_posix(), self._SLURM_USER, self._SLURM_GROUP)
+        self._CACHE_DIR.chmod(0o770)
```

Comment:
not related to the PR itself at all, but, what do you think about setting this properly? I mean, 770 means the directory is executable (?!) and any user belonging to the slurm group will be able to read this file. I think only the license manager user should be able to read this file, otherwise we kind of leak the access to the License Manager API. I believe the correct permissions here is something similar to `600`. Thoughts and comments?

Reply:
The prolog script runs as `slurm` user and the agent runs as the new `license-manager` user. This caused issues with the access for the folder and the cached token, so I included `license-manager` user to `slurm` group and changed the permission to make it accessible to both users.

Reply:
In the agent I changed it to 660, should it be the same here?

Reply:
A directory has to have the execute bit set for users to be able to access the files in it. We should constrain the permissions of the files themselves in the cache directory but leave the permissions on the cache directory wide open. I suggest 0o777 for this directory.

Reply:
why is it needed to leave the permissions wide open?